openbsd-compat/getrrsetbyname.c

```diff
@@ -170,11 +186,20 @@ struct dns_response {
 	struct dns_rr		*additional;
 };
 
+#ifndef __TANDEM
 static struct dns_response *parse_dns_response(const u_char *, int);
 static struct dns_query *parse_dns_qsection(const u_char *, int,
     const u_char **, int);
 static struct dns_rr *parse_dns_rrsection(const u_char *, int, const u_char **,
     int);
+#else
+/* NonStop compilers fail if the const is present. */
```

Comment:
can you solve this with
 #define const
at the top of the file?  we do something similar in auth-pam.c to work around API incompatibilities in PAM implementations.

Reply:
Technically, yes, but type-safety

Reply:
Technically, yes, but the OCD type-safety of mine has an issue with that.

Reply:
I meant inside an ifdef __TANDEM.  In that case the system functions that are not declared const you're not getting any type safety anyway.  This is one of the files that is synced with upstream, so extensive changes to it cause ongoing headaches.  Because of this we prefer to keep changes to such files to a minimum.

So why does the compiler choke on the const anyway?

Reply:
I'm going to be working with the compiler team on this one. It is definitely a bug - a very old one spanning at least 10 years. My hypothesis is that the symbol table has overflowed, but I cannot confirm without access to the code owned by other companies.

Reply:
More like 20 years,,,

Reply:
I will look into this particular change. It is limited in scope, so could be done.